core/src/main/java/me/mykindos/betterpvp/core/utilities/model/display/ActionBar.java

```diff
@@ -77,15 +80,32 @@ private Component nextComponent(Gamer gamer) {
                 return EMPTY;
             }
 
-            DisplayComponent display = components.peek().getRight();
+            Pair<PriorityData, DisplayComponent> peekPair = components.peek();
+            DisplayComponent display = peekPair.getRight();
             Component advComponent = display.getProvider().apply(gamer);
 
+            //peek component is not valid, try and find a valid one
+            if (advComponent == null) {
+                //components iterator/splititerator methods are not guaranteed any order, filter for one
+                Pair<PriorityData, DisplayComponent> pair = components.stream().filter(priorityDataDisplayComponentPair ->
+                        priorityDataDisplayComponentPair.getRight().getProvider().apply(gamer) != null).findFirst().orElse(null);
+
+                if (pair == null) {
+                    //there is not a valid component, return null
+                    return null;
+                }
+
+                //update display/advcomponent
+                display = pair.getRight();
+                advComponent = display.getProvider().apply(gamer);
+            }
+
+            assert (advComponent != null);
```

Comment:
It's best not to call assert in production code, I think we can just remove this and it will still behave fine?

Reply:
Use guava preconditions

Reply:
```suggestion
```